crates/sdk/src/cuda/mod.rs

```diff
@@ -77,32 +77,26 @@ impl CudaProver {
     pub fn prove<'a>(&'a self, pk: &'a SP1ProvingKey, stdin: &'a SP1Stdin) -> CudaProveBuilder<'a> {
         CudaProveBuilder { prover: self, mode: SP1ProofMode::Core, pk, stdin: stdin.clone() }
     }
-}
 
-impl Prover<CpuProverComponents> for CudaProver {
-    fn setup(&self, elf: &[u8]) -> (SP1ProvingKey, SP1VerifyingKey) {
-        let (pk, vk) = self.cuda_prover.setup(elf).unwrap();
-        (pk, vk)
-    }
-
-    fn inner(&self) -> &SP1Prover<CpuProverComponents> {
-        &self.cpu_prover
-    }
-
-    fn prove(
+    /// Proves the given program on the given input in the given proof mode.
+    ///
+    /// Returns the cycle count in addition to the proof.
+    pub fn prove_with_report(
         &self,
         pk: &SP1ProvingKey,
         stdin: &SP1Stdin,
         kind: SP1ProofMode,
-    ) -> Result<SP1ProofWithPublicValues> {
+    ) -> Result<(SP1ProofWithPublicValues, u64)> {
         // Generate the core proof.
         let proof = self.cuda_prover.prove_core(stdin)?;
```

Comment:
Can you add a note here that in a future version we could return the gas as well from `prove_core`?